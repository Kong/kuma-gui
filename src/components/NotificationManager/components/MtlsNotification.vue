<template>
  <div class="py-4">
    <p class="mb-4">
      Mutual TLS (mTLS) for communication between all the components
      of your service mesh (services, control plane, data plane proxies), proxy authentication,
      and access control rules in Traffic Permissions policies all contribute to securing your mesh.
    </p>

    <ul class="list-disc pl-4">
      <li>
        <a
          :href="`https://kuma.io/docs/${kumaDocsVersion}/security/certificates/`"
          target="_blank"
        >
          Secure access across services
        </a>
      </li>
      <li>
        <a
          :href="`https://kuma.io/docs/${kumaDocsVersion}/policies/mutual-tls/`"
          target="_blank"
        >
          Mutual TLS
        </a>
      </li>
      <li>
        <a
          :href="`https://kuma.io/docs/${kumaDocsVersion}/policies/traffic-permissions/`"
          target="_blank"
        >
          Traffic Permissions policy documentation
        </a>
      </li>
    </ul>
  </div>
</template>

<script lang="ts" setup>
import { computed } from 'vue'
import { useStore } from '@/store/store'

const store = useStore()

const kumaDocsVersion = computed(() => store.getters['config/getKumaDocsVersion'])
</script>
