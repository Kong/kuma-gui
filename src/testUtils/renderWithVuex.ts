import { render, RenderOptions, RenderResult } from '@testing-library/vue'
import merge from 'lodash/merge'
import { StoreOptions } from 'vuex'
import Vue, { ComponentOptions } from 'vue'
import Store from '@/store'

function renderWithVuex(
  Component: typeof Vue | ComponentOptions<Vue>,
  options: RenderOptions<Vue> = { store: {}, routes: [] },
): RenderResult {
  const store: StoreOptions<any> = Store()

  // Render the component and merge the original store and the custom one
  // provided as a parameter. This way, we can alter some behaviors of the
  // initial implementation.

  return render(Component, { ...options, store: merge(store, options.store), routes: options.routes })
}

export default renderWithVuex
