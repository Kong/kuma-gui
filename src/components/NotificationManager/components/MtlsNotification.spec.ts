import renderWithVuex from '@/testUtils/renderWithVuex'
import MtlsNotification from './MtlsNotification.vue'

describe('MtlsNotification.vue', () => {
  it('renders snapshot', () => {
    const { container } = renderWithVuex(MtlsNotification, {
      store: { modules: { config: { state: { version: '1.2.0' } } } },
    })

    expect(container).toMatchSnapshot()
  })
})
