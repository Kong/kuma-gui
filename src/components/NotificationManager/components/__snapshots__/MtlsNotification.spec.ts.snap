// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MtlsNotification.vue renders snapshot 1`] = `
<div>
  <div
    class="py-4"
  >
    <p
      class="mb-4"
    >
      
      Zero-trust security becomes even more critical as you transition to a microservice architecture.
      Microservices can be a chance to make your systems more secure than they were with monolithic applications.
      To do so, you use a zero-trust security model to encrypt traffic and reduce the risk of breaches,
      including mTLS and traffic permissions ACL to secure traffic paths from one service to another
    
    </p>
     
    <p
      class="mb-4"
    >
      
      Kuma is one service mesh that you can deploy once across any zone, cloud or cluster,
      including a Kubernetes cluster. Then, on top of this one service mesh implementation, you can create as many
      virtual service meshes as you wantâ€”each backed by its certificate authority.
    
    </p>
     
    <div
      class="mb-4"
    >
      <p
        class="mb-2"
      >
        
        This is the fastest and simplest way to enable mTLS in Kuma.
      
      </p>
       
      <p
        class="mb-2"
      >
        
        With a 
        <code
          class="inline"
        >
          builtin
        </code>
         CA backend type, Kuma will dynamically generate its own
        CA root certificate and key that it uses to automatically provision (and rotate) certificates for every replica
        of every service.
      
      </p>
       
      <p
        class="mb-2"
      >
        
        We can specify more than one 
        <code
          class="inline"
        >
          builtin
        </code>
         backend with different names, and each one of
        them will be automatically provisioned with a unique pair of certificate + key (they are not shared).
      
      </p>
       
      <p
        class="mb-2"
      >
        
        To enable a 
        <code
          class="inline"
        >
          builtin
        </code>
         mTLS for the entire Mesh we can apply the
        following configuration:
      
      </p>
       
      <div
        class="tab-container"
        data-testid="tab-container"
      >
        <!---->
         
        <div
          class="tab__content-container"
        >
          <div
            class="k-tabs"
            data-v-a38b664a=""
          >
            <ul
              aria-label="ktabs"
              data-v-a38b664a=""
              role="tablist"
            >
              <li
                aria-controls="panel-0"
                aria-selected="true"
                class="tab-item active"
                data-v-a38b664a=""
                id="universal-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-a38b664a=""
                >
                  Universal
                </a>
              </li>
              <li
                aria-controls="panel-1"
                aria-selected="false"
                class="tab-item"
                data-v-a38b664a=""
                id="kubernetes-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                  data-v-a38b664a=""
                >
                  Kubernetes
                </a>
              </li>
            </ul>
            <div
              aria-labelledby="universal-tab"
              class="tab-container"
              data-v-a38b664a=""
              id="panel-0"
              role="tabpanel"
              tabindex="0"
            >
              <code
                class="block"
                data-v-a38b664a=""
              >
                <pre
                  data-v-a38b664a=""
                >
                  type: Mesh
name: default
mtls:
  enabledBackend: ca-1
  backends:
    - name: ca-1
      type: builtin
      dpCert:
        rotation:
          expiration: 1d
      conf:
        caCert:
          RSAbits: 2048
          expiration: 10y
                </pre>
              </code>
              
          We can apply the configuration with
          
              <code
                class="inline"
                data-v-a38b664a=""
              >
                kumactl apply -f [..]
              </code>
              .
        
            </div>
            <div
              aria-labelledby="kubernetes-tab"
              class="tab-container"
              data-v-a38b664a=""
              id="panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
          </div>
           
          <div>
            <!---->
             
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <a
      class="k-button outline-primary"
      data-v-38fa8ecf=""
      href="https://kuma.io/docs/1.2.0/policies/mutual-tls/"
      target="_blank"
      type="button"
    >
      
      Learn More
    
    </a>
  </div>
</div>
`;
